Fixed #32681 -- Fixed VariableDoesNotExist when rendering some admin template.

mzjp2 opened the pull request:
See ticket description here: https://code.djangoproject.com/ticket/32681

Feel free to close this PR if un-needed :)

Not sure if it's worth adding a test for this - not sure where it'd go. I suspect I could do something like:

```python
self.assertEqual(ctx["subtitle"], None)
```

in `tests/admin_views.py` but not 100% sure that's useful. If it is, please let me know and I can commit that change.

felixxm commented:
@mzjp2 Thanks for this patch :+1: Please add a regression test (see for example tests added in 08c60cce3b13f6e60d7588206da2d3c71228f378) and release notes (`3.2.1.txt`). Also:
https://github.com/django/django/blob/0aacbdcf27b258387643b033352e99e6103abda8/django/contrib/admin/options.py#L1816
is not necessary anymore.

mzjp2 commented:
> @mzjp2 Thanks for this patch 👍 Please add a regression test (see for example tests added in [08c60cc](https://github.com/django/django/commit/08c60cce3b13f6e60d7588206da2d3c71228f378)) and release notes (`3.2.1.txt`). Also:
> 
> https://github.com/django/django/blob/0aacbdcf27b258387643b033352e99e6103abda8/django/contrib/admin/options.py#L1816
> 
> 
> is not necessary anymore.

Thanks for the review - I've addressed your comments in the following commits!

felixxm commented:
@mzjp2 Thanks for updates :+1: `each_context()` doesn't include a `title`, so adding a `subtitle` was inconsistent. I moved it to contexts for the affected views.

felixxm commented:
@mzjp2 Welcome aboard :boat:
